lib/Yield/Assembly/LLVMDisassemblerInterface.cpp

```diff
@@ -0,0 +1,388 @@
+                                                *InstructionInformation,
+                                                *RegisterInformation));
+  revng_assert(Printer != nullptr, "Printer object creation failed.");
+
+  using namespace options;
+  if (ImmediateStyle == ImmediateStyles::Decimal) {
+    Printer->setPrintImmHex(false);
+    Printer->setPrintBranchImmAsAddress(false);
+  } else {
+    Printer->setPrintImmHex(true);
+    Printer->setPrintBranchImmAsAddress(true);
+  }
+
+  if (ImmediateStyle == ImmediateStyles::CHexadecimal)
+    Printer->setPrintHexStyle(llvm::HexStyle::C);
+  else if (ImmediateStyle == ImmediateStyles::AsmHexadecimal)
+    Printer->setPrintHexStyle(llvm::HexStyle::Asm);
+
+  Printer->setSymbolizeOperands(ShouldSymbolizeOperands);
+  Printer->setUseMarkup(true);
+}
+
+std::pair<std::optional<llvm::MCInst>, uint64_t>
+DI::disassemble(const MetaAddress &Address,
+                llvm::ArrayRef<uint8_t> RawBytes,
+                const llvm::MCDisassembler &Disassembler) {
+  llvm::MCInst Result;
+  llvm::raw_null_ostream NullStream;
+  uint64_t LocalSize = 0;
+  auto ResultCode = Disassembler.getInstruction(Result,
+                                                LocalSize,
+                                                RawBytes,
+                                                Address.asPC(),
+                                                NullStream);
+  if (LocalSize == 0)
+    return { std::nullopt, 0 };
+
+  switch (ResultCode) {
+  case llvm::MCDisassembler::Success:
+    return std::pair{ std::move(Result), LocalSize };
+  case llvm::MCDisassembler::SoftFail:
+    return std::pair{ std::nullopt, LocalSize };
+  case llvm::MCDisassembler::Fail:
+    return std::pair{ std::nullopt, 0 };
+  default:
+    revng_abort("Unsupported MCDisassembler::getInstruction result.");
+  }
+}
+
+static assembly::Instruction::TagType parseMarkupTag(llvm::StringRef Input) {
+  if (Input == "imm:")
+    return assembly::Instruction::TagType::Immediate;
+  else if (Input == "mem:")
+    return assembly::Instruction::TagType::Memory;
+  else if (Input == "reg:")
+    return assembly::Instruction::TagType::Register;
+  else
+    revng_abort("Unknown llvm markup tag.");
+}
+
+template<typename LambdaType>
+size_t getConsecutiveCount(llvm::StringRef String,
+                           LambdaType Lambda,
+                           size_t StartFrom = 0) {
+  for (size_t Index = StartFrom; Index < String.size(); ++Index)
+    if (!Lambda(String[Index]))
+      return Index - StartFrom;
+
+  return String.size() - StartFrom;
+}
+
+template<typename LambdaType>
+size_t getBackwardsConsecutiveCount(llvm::StringRef String,
+                                    LambdaType Lambda,
+                                    size_t StartFrom) {
+  for (size_t Index = StartFrom - 1; Index != size_t(-1); --Index)
+    if (!Lambda(String[Index]))
+      return StartFrom - Index - 1;
+
+  return StartFrom;
+}
+
```

Comment:
Obscure. Docs please.

Reply:
Something like this?
https://github.com/revng/revng/pull/173/commits/805448a63b1ea6d0dbd45eccf1181b187ab4fb3e.